sdk/js/src/api/index.js

```diff
@@ -41,19 +40,17 @@ class Api {
       for (const rpcName of Object.keys(service)) {
         const rpc = service[rpcName]
 
-        this[serviceName][rpcName] = function (params = {}, body) {
-
-          const routeParams = params.route || {}
-          const queryParams = params.query || {}
+        this[serviceName][rpcName] = function ({ routeParams = {}, queryParams = {}, fieldMask = []}, payload) {
```

Comment:
why would you pass the field mask separately?
i expect it to be in the payload already in delete/put/post requests and in the query in get requests

Reply:
Exactly because of this distinction between `GET` and `POST`. Before we appended a _querified_ object directly to the url, but axios actually provides an API to do this via passing`config.params` to the request method. We should use that because it is directly integrated and also saves us a dependency (`query-string`).
I was detaching the field mask from the payload then to simplify the logic of passing this to get/non-get (either via `config.params` or `payload`) correctly.
But yeah, looking at it again, we can likely simplify this by treating it as part of the payload again and passing the payload as `config.params` in case of `GET`.